draco/tests/test_hard.py

```diff
@@ -86,3 +86,243 @@ def test_text_channel_without_text_mark():
     :- violation(_).
     """
     )
+
+
+def test_bin_and_aggregate():
+    b = hard.blocks["bin_and_aggregate"]
+    assert isinstance(b, Block)
+    p = b.program
+
+    assert is_satisfiable(
+        p
+        + """
+    entity(encoding,root,e1).
+    attribute((encoding,channel),e1,x).
+    attribute((encoding,binning),e1,10).
+    entity(encoding,root,e2).
+    attribute((encoding,channel),e2,y).
+    attribute((encoding,aggregate),e2,count).
+
+    :- violation(_).
+    """
+    )
+
+    assert not is_satisfiable(
+        p
+        + """
+    entity(encoding,root,e1).
+    attribute((encoding,channel),e1,x).
+    attribute((encoding,binning),e1,10).
+    attribute((encoding,aggregate),e1,count).
+
+    :- violation(_).
```

Comment:
I have an idea here. If we remove this line and check for violations instead of satisfiability, and test whether the list of violations is what we expect, we should have much more robust and precise tests. 

What do you think? Do you want to send a follow up pull request to make this improvement? 

```suggestion
```

Reply:
Sure. I'm going to send a new pull request for this.